File: Insertion_Sort.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}

/*
 * Complete the 'insertionSort1' function below.
 *
 * The function accepts following parameters:
 *  1. INTEGER n
 *  2. INTEGER_ARRAY arr
 */

function insertionSort1(n, arr) {
    // Write your code here
    //print the array
    let print =()=>{
    let s="";
    for(let k=0;k<n;k++){
        
        s+=(k!=n-1)?arr[k]+" " : arr[k];
        
    }
   
    console.log(s)
}

   let j=n-2;
  
   
   let e = arr[n-1];
    while(arr[j]>e){
        arr[j+1] = arr[j]
        print();
        j--;
    }
    arr[j+1] =e;
   // console.log(arr)
   print()




}

function main() {
    const n = parseInt(readLine().trim(), 10);

    const arr = readLine().replace(/\s+$/g, '').split(' ').map(arrTemp => parseInt(arrTemp, 10));

    insertionSort1(n, arr);
}

File: Insertion_Sort.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'insertionSort1' function below.
#
# The function accepts following parameters:
#  1. INTEGER n
#  2. INTEGER_ARRAY arr
#

def insertionSort1(n, arr):
  # Define the print function
  def print_arr():
      s = ""
      for k in range(n):
          s += str(arr[k]) + " " if k != n-1 else str(arr[k])
      print(s)
  
  # Call print_arr to print the initial array

  
  j = n - 2
  e = arr[n-1]
  while j >= 0 and arr[j] > e:
      arr[j+1] = arr[j]
      print_arr()
      j -= 1
  arr[j+1] = e
  print_arr()
if __name__ == '__main__':
    n = int(input().strip())

    arr = list(map(int, input().rstrip().split()))

    insertionSort1(n, arr)
